easy/validparenthesis.py

```diff
@@ -0,0 +1,98 @@
+"""
+Given a string s containing just the characters '(', ')', '{', '}', '[' and ']',
+determine if the input string is valid.
+
+An input string is valid if:
+
+Open brackets must be closed by the same type of brackets.
+Open brackets must be closed in the correct order.
+
+Examples:
+
+Input: s = "()"
+Output: true
+
+Input: s = "()[]{}"
+Output: true
+
+Input: s = "(]"
+Output: false
+
+Input: s = "([)]"
+Output: false
+
+Input: s = "{[]}"
+Output: true
+"""
+
+import unittest
+
+class Solution:
+    def isValid(self, s):
+        stack = []
+        opening_chars = ["(", "{", "["]
+        pairs = {
+            ")":"(",
+            "}":"{",
+            "]":"["
+        }
+        for element in s:
+            # if there's nothing on the stack, and s is one of
+            # (, {, [
+            if not len(stack) or element in opening_chars:
```

Comment:
Not sure about this conditional... When would you ever put an element on the stack that isn't in `opening_chars` ?